src/test/java/hudson/plugins/promoted_builds/util/ItemPathResolverTest.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2015 CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package hudson.plugins.promoted_builds.util;
+
+import edu.umd.cs.findbugs.annotations.CheckForNull;
+import hudson.model.FreeStyleProject;
+import hudson.model.Item;
+import java.io.IOException;
+import javax.annotation.Nonnull;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.MockFolder;
+
+import static org.junit.Assert.*;
+import org.junit.Before;
+
+/**
+ * Tests for {@link ItemPathResolver}.
+ * @author Oleg Nenashev
+ */
+public class ItemPathResolverTest {
+    
+    @Rule
+    public final JenkinsRule rule = new JenkinsRule();
+    
+    private MockFolder folderA, folderB, folderC;
+    private FreeStyleProject projectInTop, projectInC1, projectInC2;
+    
+    @Before
+    public void setUpFolders() throws Exception {
+        folderA = rule.createFolder("a");
+        folderB = folderA.createProject(MockFolder.class, "b");
+        folderC = folderB.createProject(MockFolder.class, "c");
+        
+        projectInTop = rule.createFreeStyleProject( "prj");
+        projectInC1 = folderC.createProject(FreeStyleProject.class, "prjInC1");
+        projectInC2 = folderC.createProject(FreeStyleProject.class, "prjInC2");
+    }
+    
+    @Test
+    public void shouldRetainTheLegacyBehavior() throws Exception {
+        assertsPath("prj", null, projectInTop);
+        
+        // FOO exists on both top level and within the folder
+        final MockFolder folder = rule.createFolder("F");
+        final FreeStyleProject prjInRoot = rule.createFreeStyleProject("FOO");
+        final FreeStyleProject prjInFolder = folder.createProject(FreeStyleProject.class, "FOO");
+        
+        // Raw addressing with different roots, should always point to the root
```

Comment:
This behavior makes no sense as it is surprising (one would assume implicit relative addressing over implicit absolute addressing), and is inconsistent with other plugins such as Copy Artifact.

I think giving relative priority over absolute, adding a compatibility warning to the plugin's pom, and documenting the behavior change on the plugin's wiki page would be the best approach here.